### ISSUE TITLE
Bring the PrefixMap in here

### ISSUE THREAD

**Issue by smashwilson** [deconst/common-python#3]
I really want to unify the `PrefixMap` implementations and parsers and bring them in here, too. I know they're a little different, but those still feel like a lot of duplicated code, and I get confused about which one I'm looking at when I'm in the other services.

**Comment by jrperritt** [comment]
True. Nearly all the code in `mapping-service` is a subset of 'layout-service`; `layout-service` just has one more level in the trie.

**Comment by smashwilson** [comment]
Yeah, they're *just* different enough to make this not trivial. I popped the issue in here mostly to remind myself to get back to this.
